src/engine/bank05.asm

```diff
@@ -4196,66 +4464,1748 @@ Func_16695: ; 16695 (5:6695)
+; - if that number is < 2  and this move is Conversion 1 OR
+; - if that number is >= 2 and this move is Conversion 2
+; then return score of $80 + 2.
+; otherwise return score of $80 + 1.
+HandlePorygonConversion: ; 16f18 (5:6f18)
+	ld a, DUELVARS_ARENA_CARD_STATUS
+	call GetTurnDuelistVariable
+	and CNF_SLP_PRZ
+	cp CONFUSED
+	jp z, HandleSpecialAIMoves.zero
+
+	ld a, [wSelectedMoveIndex]
+	or a
+	jr nz, .conversion_2
+
+; conversion 1
+	call CheckIfBenchCardsAreAtHalfHPCanEvolveAndUseSecondMove
+	cp 2
+	jr c, .low_score
+	ld a, $82
+	ret
+
+.conversion_2
+	call CheckIfBenchCardsAreAtHalfHPCanEvolveAndUseSecondMove
+	cp 2
+	jr nc, .low_score
+	ld a, $82
+	ret
+
+.low_score
+	ld a, $81
+	ret
+
+; if any Psychic Energy is found in the Discard Pile,
+; return a score of $80 + 2.
+HandleEnergyAbsorption: ; 16f41 (5:6f41)
+	ld e, PSYCHIC_ENERGY
+	ld a, CARD_LOCATION_DISCARD_PILE
+	call CheckIfAnyCardIDinLocation
+	jp nc, HandleSpecialAIMoves.zero
+	ld a, $82
+	ret
+
+; if player has cards in hand, AI calls Random:
+; - 1/3 chance to encourage move regardless;
+; - 1/3 chance to dismiss move regardless;
+; - 1/3 change to make some checks to player's hand.
+; AI tallies number of basic cards in hand, and if this
+; number is >= 2, encourage move.
+; otherwise, if it finds an evolution card in hand that
+; can evolve a card in player's deck, encourage.
+; if encouraged, returns a score of $80 + 3.
+HandleNinetalesMixUp: ; 16f4e (5:6f4e)
+	ld a, DUELVARS_NUMBER_OF_CARDS_IN_HAND
+	call GetNonTurnDuelistVariable
+	or a
+	ret z ; return if no hand cards
+
+	ld a, 3
+	call Random
+	or a
+	jr z, .encourage ; if a = 0
+	dec a
+	ret z ; return if a = 1
+	call SwapTurn
+	call CreateHandCardList
+	call SwapTurn
+	or a
+	ret z ; return if no hand cards (again)
+	ld a, DUELVARS_NUMBER_OF_POKEMON_IN_PLAY_AREA
+	call GetNonTurnDuelistVariable
+	cp 3
+	jr nc, .check_play_area
+
+	ld hl, wDuelTempList
+	ld b, 0
+.loop_hand
+	ld a, [hli]
+	cp $ff
+	jr z, .tally_basic_cards
+	push bc
+	call SwapTurn
+	call LoadCardDataToBuffer2_FromDeckIndex
+	call SwapTurn
+	pop bc
+	ld a, [wLoadedCard2Type]
+	cp TYPE_ENERGY
+	jr nc, .loop_hand
+	ld a, [wLoadedCard2Stage]
+	or a
+	jr nz, .loop_hand
+	; is a basic Pokémon card
+	inc b 
+	jr .loop_hand
+.tally_basic_cards
+	ld a, b
+	cp 2
+	jr nc, .encourage
+
+; less than 2 basic cards in hand
+.check_play_area
+	ld a, DUELVARS_ARENA_CARD
+	call GetNonTurnDuelistVariable
+.loop_play_area
+	ld a, [hli]
+	cp $ff
+	jp z, HandleSpecialAIMoves.zero
+	push hl
+	call SwapTurn
+	call CheckForEvolutionInList
+	call SwapTurn
+	pop hl
+	jr nc, .loop_play_area
+
+.encourage
+	ld a, $83
+	ret
+
+; return score of $80 + 3.
+HandleZapdos3BigThunder: ; 16fb8 (5:6fb8)
+	ld a, $83
+	ret
+
+; dismiss move if cards in deck <= 20.
+; otherwise return a score of $80 + 0.
+HandleKangaskhanFetch: ; 16fbb (5:6fbb)
+	ld a, DUELVARS_NUMBER_OF_CARDS_NOT_IN_DECK
+	call GetTurnDuelistVariable
+	cp 41
```

Comment:
would something like DECK_AMT + 1 - 20 be better here? I don't actually see a constant for deck size though, even with a bunch of hex values equaling 60 in bank 2. This might be something to look at another time.